doc/designs/quic-design/connection-state-machine.md

```diff
@@ -0,0 +1,637 @@
+</tr>
+<tr>
+  <td rowspan="3"><tt>ACTIVE.ESTABLISHING.INITIAL_EXCHANGE_B</tt></td>
+  <td rowspan="3"><tt>enter:SendPackets()</tt><br/>
+    (First Initial, with token)<br/>
+    (*All further Initial packets contain the token)<br/>(*PN is not reset)</td>
+  <td>—<tt>RX:INITIAL</tt>→</td>
+  <td><tt>ACTIVE.ESTABLISHING.INITIAL_EXCHANGE_CONTINUED</tt></td>
+</tr>
+<tr>
+  <td>—<tt>PROBE_TIMEOUT</tt>→</td>
+  <td>TODO: Tail loss probe for initial packets?</td>
+</tr>
+<tr>
+  <td>—<tt>CAN_SEND</tt>→</td>
+  <td><tt>SendPackets()</tt></td>
+</tr>
+<tr>
+  <td rowspan="2"><tt>ACTIVE.ESTABLISHING.INITIAL_EXCHANGE_CONTINUED</tt></td>
+  <td rowspan="2"><tt>enter:SendPackets()</tt></td>
+  <td>—<tt>RX:INITIAL</tt>→</td>
+  <td>(packet processed, no change)</td>
+</tr>
+<tr>
+  <td>—<tt>TLS:HAVE_EL(HANDSHAKE)</tt>→</td>
+  <td><tt>ACTIVE.ESTABLISHING.HANDSHAKE</tt></td>
+</tr>
+<tr>
+  <td rowspan="3"><tt>ACTIVE.ESTABLISHING.HANDSHAKE</tt></td>
+  <td rowspan="3"><tt>enter:ProvisionEL(Handshake)</tt><br/>
+  <tt>enter:SendPackets()</tt> (First Handshake packet, if pending)</td>
+  <td>—<tt>RX:HANDSHAKE</tt>→</td>
+  <td><tt>ACTIVE.ESTABLISHING.HANDSHAKE_CONTINUED</tt></td>
+</tr>
+<tr>
+  <td>—<tt>RX:INITIAL</tt>→</td>
+  <td>(packet processed if EL is not dropped)</td>
+</tr>
+<tr>
+  <td>—<tt>CAN_SEND</tt>→</td>
+  <td><tt>SendPackets()</tt></td>
+</tr>
+<tr>
+  <td rowspan="3"><tt>ACTIVE.ESTABLISHING.HANDSHAKE_CONTINUED</tt></td>
+  <td rowspan="3"><tt>enter:DropEL(Initial)</tt><br/><tt>enter:SendPackets()</tt></td>
+  <td>—<tt>RX:HANDSHAKE</tt>→</td>
+  <td>(packet processed, no change)</td>
+</tr>
+<tr>
+  <td>—<tt>TLS:HANDSHAKE_COMPLETE</tt>→</td>
+  <td><tt>ACTIVE.ESTABLISHING.HANDSHAKE_COMPLETE</tt></td>
+</tr>
+ <tr>
+  <td>—<tt>CAN_SEND</tt>→</td>
+  <td><tt>SendPackets()</tt></td>
+</tr>
+<tr>
+  <td rowspan="3"><tt>ACTIVE.ESTABLISHING.HANDSHAKE_COMPLETED</tt></td>
+  <td rowspan="3"><tt>enter:ProvisionEL(1RTT)</tt><br/><tt>enter:HandshakeComplete()</tt><br/><tt>enter[server]:Send(HANDSHAKE_DONE)</tt><br/><tt>enter:SendPackets()</tt></td>
+  <td>—<tt>RX:1RTT[HANDSHAKE_DONE]</tt>→</td>
+  <td><tt>ACTIVE.ESTABLISHING.HANDSHAKE_CONFIRMED</tt></td>
+</tr>
+<tr>
+  <td>—<tt>RX:1RTT</tt>→</td>
+  <td>(packet processed, no change)</td>
+</tr>
+ <tr>
+  <td>—<tt>CAN_SEND</tt>→</td>
+  <td><tt>SendPackets()</tt></td>
+</tr>
+<tr>
+  <td rowspan="1"><tt>ACTIVE.ESTABLISHING.HANDSHAKE_CONFIRMED</tt></td>
+  <td rowspan="1"><tt>enter:DiscardEL(Handshake)</tt><br/><tt>enter:Permit1RTTKeyUpdate()</tt></td>
+  <td>—ε→</td>
+  <td><tt>ACTIVE.OPEN</tt></td>
+</tr>
+<tr>
+  <td rowspan="2"><tt>ACTIVE.OPEN</tt></td>
+  <td rowspan="2"></td>
+  <td>—<tt>RX:1RTT</tt>→</td>
+  <td>(packet processed, no change)</td>
+</tr>
+<tr>
+  <td>—<tt>CAN_SEND</tt>→</td>
+  <td><tt>SendPackets()</tt></td>
+</tr>
+<tr>
+  <td><tt>TERMINATING</tt></td>
+  <td></td>
+  <td>—<tt>TERMINATING_TIMEOUT</tt>→</td>
+  <td><tt>TERMINATED</tt></td>
```

Comment:
The diagram shows 2 events leading from the "TERMINATING" state (TERMINATING_TIMEOUT and RX:STATELESS_RESET) however this table only shows TERMINATING_TIMEOUT?